"retry: false" of "http>:" operator is not working

When a url is wrong or temporarily down, the `http>:` operator keeps retrying even if `retry: false` is specified. 

Digdag version is 0.9.6. A sample digdag workflow is as following.

```
+check:
  +call:
    _error:
      fail>: "Error!"

    +call:
      http>: http://wwwW.xxxx/v1/stats/qps
      retry: false
      store_content: true
```

And digdag server log is as follows.

```
2017-03-15 18:18:54 +0900 [INFO] (0785@+test+check+call+call) io.digdag.core.agent.OperatorManager: http>: http://wwww.xxxx/v1/stats/qps
2017-03-15 18:18:54 +0900 [INFO] (0785@+test+check+call+call) io.digdag.standards.operator.HttpOperatorFactory$HttpOperator: Sending HTTP request: GET http://wwww.xxxx/v1/stats/qps
2017-03-15 18:18:54 +0900 [WARN] (0785@+test+check+call+call) io.digdag.standards.operator.state.PollingRetryExecutor: HTTP request failed: retrying in 1 seconds
java.lang.RuntimeException: java.net.UnknownHostException: wwww.xxxx: Name or service not known
    at com.google.api.client.repackaged.com.google.common.base.Throwables.propagate(Throwables.java:160)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.send(HttpOperatorFactory.java:324)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.execute(HttpOperatorFactory.java:250)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.lambda$run$1(HttpOperatorFactory.java:195)
    at io.digdag.standards.operator.state.PollingRetryExecutor.run(PollingRetryExecutor.java:176)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.run(HttpOperatorFactory.java:195)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.runTask(HttpOperatorFactory.java:136)
    at io.digdag.util.BaseOperator.run(BaseOperator.java:35)
    at io.digdag.core.agent.OperatorManager.callExecutor(OperatorManager.java:314)
    at io.digdag.core.agent.OperatorManager.runWithWorkspace(OperatorManager.java:255)
    at io.digdag.core.agent.OperatorManager.lambda$runWithHeartbeat$2(OperatorManager.java:138)
    at io.digdag.core.agent.ExtractArchiveWorkspaceManager.withExtractedArchive(ExtractArchiveWorkspaceManager.java:36)
    at io.digdag.core.agent.OperatorManager.runWithHeartbeat(OperatorManager.java:136)
    at io.digdag.core.agent.OperatorManager.run(OperatorManager.java:120)
    at io.digdag.core.agent.MultiThreadAgent.lambda$run$0(MultiThreadAgent.java:121)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.UnknownHostException: wwww.xxxx: Name or service not known
    at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
    at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:928)
    at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1323)
    at java.net.InetAddress.getAllByName0(InetAddress.java:1276)
    at java.net.InetAddress.getAllByName(InetAddress.java:1192)
    at java.net.InetAddress.getAllByName(InetAddress.java:1126)
    at org.eclipse.jetty.util.SocketAddressResolver$Async.lambda$resolve$1(SocketAddressResolver.java:167)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
    ... 1 common frames omitted
2017-03-15 18:18:56 +0900 [INFO] (0785@+test+check+call+call) io.digdag.core.agent.OperatorManager: http>: http://wwww.xxxx/v1/stats/qps
2017-03-15 18:18:56 +0900 [INFO] (0785@+test+check+call+call) io.digdag.standards.operator.HttpOperatorFactory$HttpOperator: Sending HTTP request: GET http://wwww.xxxx/v1/stats/qps
2017-03-15 18:18:56 +0900 [WARN] (0785@+test+check+call+call) io.digdag.standards.operator.state.PollingRetryExecutor: HTTP request failed: retrying in 2 seconds
java.lang.RuntimeException: java.net.UnknownHostException: wwww.xxxx
    at com.google.api.client.repackaged.com.google.common.base.Throwables.propagate(Throwables.java:160)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.send(HttpOperatorFactory.java:324)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.execute(HttpOperatorFactory.java:250)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.lambda$run$1(HttpOperatorFactory.java:195)
    at io.digdag.standards.operator.state.PollingRetryExecutor.run(PollingRetryExecutor.java:176)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.run(HttpOperatorFactory.java:195)
    at io.digdag.standards.operator.HttpOperatorFactory$HttpOperator.runTask(HttpOperatorFactory.java:136)
    at io.digdag.util.BaseOperator.run(BaseOperator.java:35)
    at io.digdag.core.agent.OperatorManager.callExecutor(OperatorManager.java:314)
    at io.digdag.core.agent.OperatorManager.runWithWorkspace(OperatorManager.java:255)
    at io.digdag.core.agent.OperatorManager.lambda$runWithHeartbeat$2(OperatorManager.java:138)
    at io.digdag.core.agent.ExtractArchiveWorkspaceManager.withExtractedArchive(ExtractArchiveWorkspaceManager.java:36)
    at io.digdag.core.agent.OperatorManager.runWithHeartbeat(OperatorManager.java:136)
    at io.digdag.core.agent.OperatorManager.run(OperatorManager.java:120)
    at io.digdag.core.agent.MultiThreadAgent.lambda$run$0(MultiThreadAgent.java:121)
...

(repeating...)

```

--- a/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
+++ b/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
@@ -255,12 +255,12 @@ public class HttpOperatorFactory
                 throw error(req, uriIsSecret, e.getResponse());
             }
             catch (RuntimeException e) {
-                logger.info("Exception without response: {} {}", req.getMethod(), safeUri);
-                if(retry){
+                logger.warn("Exception without response: {} {}", req.getMethod(), safeUri);
+                if (retry) {
                     throw e;
                 }
                 else {
-                    throw new TaskExecutionException(e.toString());
+                    throw new TaskExecutionException(e);
                 }
             }
 

--- a/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
+++ b/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
@@ -257,7 +257,7 @@ public class HttpOperatorFactory
             catch (RuntimeException e) {
                 logger.info("Exception without response: {} {}", req.getMethod(), safeUri);
                 if(retry){
-                    throw new RuntimeException(e.toString());
+                    throw e;
                 }
                 else {
                     throw new TaskExecutionException(e.toString());

--- a/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
+++ b/digdag-standards/src/main/java/io/digdag/standards/operator/HttpOperatorFactory.java
@@ -254,6 +254,15 @@ public class HttpOperatorFactory
             catch (HttpResponseException e) {
                 throw error(req, uriIsSecret, e.getResponse());
             }
+            catch (RuntimeException e) {
+                logger.info("Exception without response: {} {}", req.getMethod(), safeUri);
+                if(retry){
+                    throw new RuntimeException(e.toString());
+                }
+                else {
+                    throw new TaskExecutionException(e.toString());
+                }
+            }
 
             logger.info("Received HTTP response: {} {}: {}", req.getMethod(), safeUri, res);
 

